packages/flutter/lib/src/material/navigation_bar_theme.dart

```diff
@@ -0,0 +1,221 @@
+      indicatorColor: Color.lerp(a?.indicatorColor, b?.indicatorColor, t),
+      labelTextStyle: _lerpProperties<TextStyle?>(a?.labelTextStyle, b?.labelTextStyle, t, TextStyle.lerp),
+      iconTheme: _lerpProperties<IconThemeData?>(a?.iconTheme, b?.iconTheme, t, IconThemeData.lerp),
+      labelBehavior: t < 0.5 ? a?.labelBehavior : b?.labelBehavior,
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      height,
+      backgroundColor,
+      indicatorColor,
+      labelTextStyle,
+      iconTheme,
+      labelBehavior,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is NavigationBarThemeData
+        && other.height == height
+        && other.backgroundColor == backgroundColor
+        && other.indicatorColor == indicatorColor
+        && other.labelTextStyle == labelTextStyle
+        && other.iconTheme == iconTheme
+        && other.labelBehavior == labelBehavior;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('height', height, defaultValue: null));
+    properties.add(ColorProperty('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(ColorProperty('indicatorColor', indicatorColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<TextStyle?>>('labelTextStyle', labelTextStyle, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<IconThemeData?>>('iconTheme', iconTheme, defaultValue: null));
+    properties.add(DiagnosticsProperty<NavigationDestinationLabelBehavior>('labelBehavior', labelBehavior, defaultValue: null));
+  }
+
+  static MaterialStateProperty<T>? _lerpProperties<T>(
+      MaterialStateProperty<T>? a,
+      MaterialStateProperty<T>? b,
+      double t,
+      T Function(T?, T?, double) lerpFunction,
+      ) {
+    // Avoid creating a _LerpProperties object for a common case.
+    if (a == null && b == null)
+      return null;
+    return _LerpProperties<T>(a, b, t, lerpFunction);
+  }
+}
+
+class _LerpProperties<T> implements MaterialStateProperty<T> {
+  const _LerpProperties(this.a, this.b, this.t, this.lerpFunction);
+
+  final MaterialStateProperty<T>? a;
+  final MaterialStateProperty<T>? b;
+  final double t;
+  final T Function(T?, T?, double) lerpFunction;
+
+  @override
+  T resolve(Set<MaterialState> states) {
+    final T? resolvedA = a?.resolve(states);
+    final T? resolvedB = b?.resolve(states);
+    return lerpFunction(resolvedA, resolvedB, t);
+  }
+}
+
+/// An inherited widget that defines visual properties for [NavigationBar]s and
+/// [NavigationDestination]s in this widget's subtree.
+///
+/// Values specified here are used for [NavigationBar] properties that are not
+/// given an explicit non-null value.
```

Comment:
It would be helpful to include a see-also link to ThemeData.navigationBarTheme.

Reply:
Done